src/widgets/pagination/__tests__/pagination-test.js

```diff
@@ -136,3 +136,49 @@ describe('pagination()', () => {
     };
   }
 });
+
+describe('pagination MaxPage', () => {
+  jsdom({useEach: true});
+
+  let ReactDOM;
+  let container;
+  let widget;
+  let results;
+  let cssClasses;
+
+  beforeEach(() => {
+    ReactDOM = {render: sinon.spy()};
+    pagination.__Rewire__('ReactDOM', ReactDOM);
+    pagination.__Rewire__('autoHideContainerHOC', sinon.stub().returns(Pagination));
+
+    container = document.createElement('div');
+    cssClasses = {
+      root: 'root',
+      item: 'item',
+      link: 'link',
+      page: 'page',
+      previous: 'previous',
+      next: 'next',
+      first: 'first',
+      last: 'last',
+      active: 'active',
+      disabled: 'disabled'
+    };
+    results = {hits: [{first: 'hit', second: 'hit'}], nbHits: 300, hitsPerPage: 10, nbPages: 30};
+  });
+
+  it('does to have any default', () => {
+    widget = pagination({container, scrollTo: false, cssClasses});
+    expect(widget.getMaxPage(results)).toEqual(30);
+  });
+
+  it('does reduce the number of page if lower than nbPages', () => {
+    widget = pagination({container, scrollTo: false, cssClasses, maxPages: 20});
+    expect(widget.getMaxPage(results)).toEqual(20);
+  });
```

Comment:
This is much more readable than before, thanks! 

I still find tests are even more readable when all that is needed for it is explicitly set in the test itself. Here, part of the knowledge is in the `beforeEach` (the fact that we set `nbPages: 30`).

What about putting a `paginationOptions = {container, scrollTo: false, cssClasses}` in the `beforeEach`, and then writing tests like:

```
// Given
paginationOptions.maxPages = 20;
results.nbPages = 30;

// When
var actual = pagination(paginationOptions).getMaxPage(results);

// Then
expect(actual).toEqual(20);
```

I wouldn't say this is blocking though, I just find it easier to understand the tests that way. We split it in the initial test state, the action, and the expectation. Any thoughts @vvo?